packages/astro/src/assets/build/generate.ts

```diff
@@ -57,6 +58,23 @@ export async function generateImage(
 	const finalFileURL = new URL('.' + filepath, clientRoot);
 	const finalFolderURL = new URL('./', finalFileURL);
 
+	// The original filepath or URL from the image transform
+	const originalImagePath = isLocalImage
+		? (options.src as ImageMetadata).src
+		: (options.src as string);
+
+	if (!ssr && config.image.removeOriginals) {
+		const originalFileURL = new URL('.' + originalImagePath, clientRoot);
+		try {
+			await fs.promises.unlink(originalFileURL);
+		} catch (e) {
+			logger.warn(
+				'astro:assets',
+				`An error was encountered while removing asset original. Error: ${e}`
+			);
+		}
+	}
```

Comment:
This needs to be done outside of this function, as you might have multiple optimized versions of a same asset. I'd suggest having a Set that you add to and then loop through to remove the files

Reply:
@Princesseuh, what about fill the Set inside this loop and place new code below?

https://github.com/withastro/astro/blob/main/packages/astro/src/core/build/generate.ts#L200C1-L202

Reply:
Yes, that would be fine!

Reply:
@Princesseuh, not sure I done this pretty, but I'm done 🙂